RocketApp/Features/Sources/RocketLaunch/RocketLaunchView.swift

```diff
@@ -1,13 +1,221 @@
+    self.store = store
+    self.viewStore = ViewStore(store)
+  }
+
+  public var body: some View {
+    VStack(spacing: 0) {
+      Spacer()
+
+      HStack {
+        Spacer()
+          .frame(maxWidth: viewStore.lWidth)
+
+        // TODO: All view logic move to ViewState.
+        if viewStore.rocketHasLaunched {
+          flyingRocket
+        } else {
+          Image.rocketIdle
+        }
+
+        Spacer()
+          .frame(maxWidth: viewStore.rWidth)
+      }
+
+      Spacer()
+        .frame(maxHeight: max(viewStore.height, 0))
+
+      platform
+    }
+    .frame(height: UIScreen.main.bounds.height)
+    .ignoresSafeArea()
+    .background {
+      ZStack {
+        space
+
+        daySky
+      }
+    }
+    .overlay { flash }
+    .overlay {
+      if let error = viewStore.motionError {
+        errorView(error)
+      }
+    }
+    .onAppear { viewStore.send(.onAppear) }
+  }
+
+  private var flyingRocket: some View {
+    Image.rocketFlying.opacity(2)
+      .scaleEffect(rocketScale)
+      .overlay(alignment: .bottom) {
+        Group {
+          HStack {
+            ForEach((1...3), id: \.self) { _ in
+              Circle()
+                .trim(from: 0, to: 0.6)
+                .stroke()
+            }
+          }
+          .frame(width: 100, height: 100)
+
+          HStack {
+            ForEach((1...2), id: \.self) { _ in
+              Circle()
+                .trim(from: 0, to: 0.6)
+                .stroke()
+            }
+          }
+          .frame(width: 80, height: 100)
+        }
+        .offset(y: 50)
+        .foregroundColor(.gray)
+        .scaleEffect(pollutionScale)
+        .onAppear {
+          withAnimation(
+            Animation.easeInOut(duration: 0.08)
+              .repeatForever(autoreverses: true)
+          ) {
+            rocketScale = 0.99
+            pollutionScale = 1
+          }
+        }
+        .onDisappear {
+          rocketScale = 1
+          pollutionScale = 0.5
+        }
+      }
+  }
+
+  @ViewBuilder
+  private var platform: some View {
+    ZStack(alignment: .top) {
+      Rectangle()
+        .fill(.indigo)
+        .frame(maxWidth: .infinity)
+        .ignoresSafeArea()
+        .frame(height: 100)
+        .opacity(viewStore.rocketHasLaunched ? 0 : 1)
+
+      VStack {
+        HStack {
+          Spacer()
+
+          Text(viewStore.rocketHasLaunched ? .launchSuccessful : .tiltToLaunch(viewStore.rocketData.name))
+            .font(.headline)
+            .multilineTextAlignment(.center)
+            .bold()
+            .foregroundColor(viewStore.rocketHasLaunched ? .indigo : .white)
+            .opacity(1 - backgroundOpacity)
+
+          Spacer()
+
+          if !viewStore.rocketHasLaunched {
+            launchCounter
+          }
+        }
+        .padding([.top, .trailing, .leading])
+
+        if viewStore.rocketHasLaunched
+            && viewStore.neededTiltToLaunch - viewStore.calculatedHeight > viewStore.neededTiltToLaunch {
```

Comment:
Nebavili sme se o tom, ze to `&&` budem davat na konec radky a ne na zacatek? Nebo si to pletu?

Reply:
Asi bavili. Je to skutecne obecna konvence? Je prehlednejsi kdyz tyhle spojky pisu k prislusnym vyrazum. Stejne jako se pisou spojky ve vetach na zacatek radky k prislusnym podstatnym jmenum.

Reply:
`&&` se pisou na konci radky. Pokud to nekdo na projektech ma nastaveny jinak, tak budiz, ale odjakziva co programuju se to pise na konci radky.

Reply:
Ok.